src/test/java/htsjdk/variant/vcf/AbstractVCFCodecTest.java

```diff
@@ -66,4 +67,30 @@ public void testGLnotOverridePL() {
         }
         Assert.assertEquals(variant.getGenotype(0).getPL(), new int[]{45, 0, 50});
     }
+
+    @DataProvider(name = "caseIntolerantDoubles")
+    public Object[][] getCaseIntolerantDoubles() {
+        return new Object[][]{
+                {"src/test/resources/htsjdk/variant/test_withNanQual.vcf", Double.NaN},
+                {"src/test/resources/htsjdk/variant/test_withPosInfQual.vcf", Double.POSITIVE_INFINITY},
+                {"src/test/resources/htsjdk/variant/test_withNegInfQual.vcf", Double.NEGATIVE_INFINITY},
+        };
+    }
+
+    @Test(dataProvider = "caseIntolerantDoubles")
+    public void testCaseIntolerantDoubles(String vcfInput, Double value) {
```

Comment:
unnecessary to use a boxed type here as `value` is never `null`
```suggestion
    public void testCaseIntolerantDoubles(String vcfInput, double value) {
```